File: cpp/Kode/Type.cpp
#include "Type.h"

Type::Type(std::string dmg, std::string atk, std::string scl) : 
    DamageType(dmg), AttackType(atk), Scaling(scl) {}

void Type::displayType() const {
    std::cout << "[TYPE] Damage: " << DamageType << ", Attack: " << AttackType << ", Scaling: " << Scaling;
}

File: java/Kode/Type.java
public class Type {
    private String damageType;
    private String attackType;
    private String scaling;

    public Type(String damageType, String attackType, String scaling) {
        this.damageType = damageType;
        this.attackType = attackType;
        this.scaling = scaling;
    }

    public String getDamageType() {
        return damageType;
    }

    public void displayType() {
        System.out.print("[TYPE] Damage: " + damageType + ", Attack: " + attackType + ", Scaling: " + scaling);
    }
}